base/tps/src/org/dogtagpki/server/tps/rest/ProfileService.java

```diff
@@ -94,30 +102,51 @@ public Response findProfiles(String filter, Integer start, Integer size) {
             throw new BadRequestException("Filter is too short.");
         }
 
-        start = start == null ? 0 : start;
-        size = size == null ? DEFAULT_SIZE : size;
-
+        CMS.debug("ProfileService.j.findProfiles filter: " + filter);
         try {
+            List<String> authorizedProfiles = getAuthorizedProfiles();
+
+            start = start == null ? 0 : start;
+            size = size == null ? DEFAULT_SIZE : size;
+
             TPSSubsystem subsystem = (TPSSubsystem) CMS.getSubsystem(TPSSubsystem.ID);
             ProfileDatabase database = subsystem.getProfileDatabase();
 
-            Iterator<ProfileRecord> profiles = database.findRecords(filter).iterator();
+            Collection<ProfileRecord> profiles = new ArrayList<>();
```

Comment:
It should be possible to use a `List` here too to avoid using the `Iterator`, but the code should work as is:
```
List<ProfileRecord> profiles = new ArrayList<>();
```